### DISCUSSION TITLE
Importing code chunks from an external R script

### DISCUSSION THREAD

**Discussion by arnaudgallou** [quarto-dev/quarto-cli#1005] [Q&A] [answered]
Is there a way to import chunks of a `R` script in a `.qmd` document? I have tried using `knitr::read_chunk()` (see example below or [here](https://github.com/arnaudgallou/dummy-qmd/tree/main/read_chunk)) but I didn't manage to make it work in Quarto.

`read_chunk.R`:
```
x <- "blah"

# ---- foo ----

ggplot() +
  geom_segment(aes(x = 0, xend = 1, y = 0, yend = 1)) +
  lims(x = c(0, 1), y = c(0, 1)) +
  theme(
    text = element_text(size = 20),
    axis.ticks = element_blank(),
    axis.text = element_blank()
  ) +
  ggtitle("foo")

# ---- bar ----

ggplot() +
  geom_segment(aes(x = 1, xend = 0, y = 1, yend = 0)) +
  lims(x = c(0, 1), y = c(0, 1)) +
  theme(
    text = element_text(size = 20),
    axis.ticks = element_blank(),
    axis.text = element_blank()
  ) +
  ggtitle("bar")
```

`read_chunk.qmd`:
``````
---
title: Importing code chunk using `knitr::read_chunk()`
editor_options:
  markdown: 
    wrap: sentence
execute:
  echo: false
  warning: false
---

```{r}
library(ggplot2)
knitr::read_chunk("read_chunk.R")
```

```{r foo}
```
``````

Did I do something wrong in my example? If not, is there a similar tool or feature available in Quarto to do that?

Please note that converting the `.qmd` to `.Rmd` will correctly import and render the `foo` chunk.

**Comment by jjallaire** [comment]
@yihui I've confirmed that the call to `read_chunk()` doesn't seem to work in quarto. Could this be due to some of our hooks? Perhaps there is a line or two of code we need to add in order to respect cells populated via labels from `read_chunk()` ?

**Reply by yihui** [reply]
I don't know and don't have time to debug it until I come back from vacation. You can check if `knitr::knit_code$get()` contains the code after calling `read_chunk()`. Then figure out why the code is not applied to the chunk.

**Comment by jjallaire** [comment]
I spent some time trying to figure out what was going on -- it seems like perhaps in the case of quarto the `code` option is explicitly set somehow (although I couldn't see where this might be happening in either quarto or knitr::parse_block). If `code` is in fact set then this will override any code read via `read_chunk()`

**Reply by arnaudgallou** [reply]
Thank you for looking into the problem. I will use a different approach for now. I hope a solution that works will be implemented in the future.

**Reply by jjallaire** [reply]
Thanks again for reporting this, should be fixed here: https://github.com/quarto-dev/quarto-cli/commit/a120c2a2ddf17d71447315efb6694237545ccf52

**Reply by arnaudgallou** [reply]
Works like a charm. Thank you.

**Comment by cderv** [comment, marked as answer]
This was related to other issue (https://github.com/quarto-dev/quarto-cli/issues/1034) and now fixed in Quarto v0.9.499 and above, as mentioned [above](https://github.com/quarto-dev/quarto-cli/discussions/1005#discussioncomment-2867682)
